### PR TITLE
Create 07-special-density.rmd

### PR DESCRIPTION
**Pull request by RaphaelS1**
This will error until https://github.com/mlr-org/mlr3viz/pull/25 is merged

### CODE REVIEW

File: bookdown/07-special-density.rmd [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+head(task$truth())
+
+# histogram and density plot
+library("mlr3viz")
+autoplot(task, type = "overlay")
+```
+
+Unconditional density estimation is an unsupervised method. Hence, `TaskDens` is an unsupervised task which inherits directly from `Task` unlike `TaskClassif` and `TaskRegr`. However, `TaskDens` still has a `target` and a `$truth` field defined by:
+  
+  * `target` - the variable for which to estimate density
```
Comment (mllg) [inline bookdown/07-special-density.rmd]: Is it `target` or `target_names`?

File: bookdown/07-special-density.rmd [hunk 2]
```diff
@@ -0,0 +1,89 @@
+## Density Estimation {#density}
```
Comment (mllg) [inline bookdown/07-special-density.rmd]: Can you rename the file extension to `Rmd` (with an uppercase R)?

File: bookdown/07-special-density.rmd [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+Density estimation is the learning task to find the unknown distribution from which an i.i.d. data set is generated. We interpret this broadly, with this distribution not necessarily being continuous (so may possess a mass not density). The conditional case, where a distribution is predicted conditional on covariates, is known as ‘probabilistic supervised regression’, and will be implemented in `r mlr_pkg("mlr3proba")` in the near-future. Unconditional density estimation is viewed as an unsupervised task.
```
Comment (mllg) [inline bookdown/07-special-density.rmd]: Nitpicking here, but we agreed on trying to have one sentence per line. This greatly simplifies reading of diffs.

File: bookdown/07-special-density.rmd [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+# histogram and density plot
+library("mlr3viz")
+autoplot(task, type = "overlay")
+```
+
+Unconditional density estimation is an unsupervised method. Hence, `TaskDens` is an unsupervised task which inherits directly from `Task` unlike `TaskClassif` and `TaskRegr`. However, `TaskDens` still has a `target` and a `$truth` field defined by:
+  
+  * `target` - the variable for which to estimate density
+  * `truth` -  the `target`. This is *not* the true density which is always unknown.
```
Comment (mllg) [inline bookdown/07-special-density.rmd]: `truth` - the values of the target column.

?

File: bookdown/07-special-density.rmd [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+```
+
+Unconditional density estimation is an unsupervised method. Hence, `TaskDens` is an unsupervised task which inherits directly from `Task` unlike `TaskClassif` and `TaskRegr`. However, `TaskDens` still has a `target` and a `$truth` field defined by:
+  
+  * `target` - the variable for which to estimate density
+  * `truth` -  the `target`. This is *not* the true density which is always unknown. 
+  
+### Train and Predict
+
+Density learners have `train` and `predict` methods, though being unsupervised, 'prediction' is actually 'estimation'. In training, a [distr6](https://cran.r-project.org/package=distr6) object is created,
```
Comment (mllg) [inline bookdown/07-special-density.rmd]: We have `r cran_pkg("distr6")` to create hyperrefs to cran.

File: bookdown/07-special-density.rmd [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+[see here](https://alan-turing-institute.github.io/distr6/) for full tutorials on how to access the `pdf`, `cdf`, and other important fields and methods. The predict method is simply a wrapper around `self$model$pdf` and if available `self$model$cdf`, i.e. evaluates the pdf/cdf at given points. Note
```
Comment (mllg) [inline bookdown/07-special-density.rmd]: Maybe write down what `pdf` and `cdf` stand for on its first use?

File: bookdown/07-special-density.rmd [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+# Logloss for probabilistic predictions
+measure = msr("dens.logloss")
+print(measure)
+
+set.seed(1)
+bmr = benchmark(benchmark_grid(task, learners, rsmp("cv", folds = 3)))
+bmr$aggregate(measure)
+autoplot(bmr, measure = measure)
+```
```
Comment (mllg) [inline bookdown/07-special-density.rmd]: Can you summarize the (probably boring?) results with 1-2 sentences?

File: bookdown/07-special-density.rmd (hunk 3 above)
Comment (mllg) [reply bookdown/07-special-density.rmd]: Also makes it possible to comment on sentences instead of paragraphs. ;)

File: bookdown/07-special-density.rmd (hunk 6 above)
Comment (mllg) [reply bookdown/07-special-density.rmd]: Also, if there is a good introduction to the topic on wikipedia or in a text book, please reference it.

File: bookdown/07-special-density.rmd (hunk 4 above)
Comment (RaphaelS1) [reply bookdown/07-special-density.rmd]: `truth` -  the values of the `target` column (which is *not* the true density which is always unknown)

?

Comment (mllg) [reply bookdown/07-special-density.rmd]: Yes, this is what I meant. I just wanted it to include the word "values".

**Comment by RaphaelS1** [comment]
Thanks, changes made as suggested. I'm not 100% sure I added the reference correctly and will take a look if this builds properly. I'll also make the relevant changes suggested here to the survival sub-chapter as well.

**Comment by mllg** [comment]
Ready to be merged?

**Comment by RaphaelS1** [comment]
> Ready to be merged?

From my side, yes
